form.js

```diff
@@ -0,0 +1,296 @@
+import { everyErrorText } from "./every_error_text.js";
+import { RegEx } from "./RegEx.js";
+import { APIweather } from "./fetch_API_Weather.js";
+import { APIerrors } from "./APIErrors.js";
+
+const controlButton = document.querySelector(".control-button");
+const form = document.querySelector("form");
+const endForm = document.querySelector(".end-form");
+const endFormText = document.querySelector(".end-form-text");
+const styleInputInformation = document.querySelectorAll(
+	".style-input-information"
+);
+const problemWithServerOrCode = document.querySelector(
+	".problem-with-server-or-code"
+);
+
+const allGoodAnswerIcon = document.querySelectorAll(".fa-check");
+const allWrongAnswerIcon = document.querySelectorAll(".fa-x");
+
+const goNextBtn = document.querySelector(".go-next-btn");
+const clearFormBtn = document.querySelector(".clear-form-btn");
+
+const theMinimumNumberOfCharactersInTheNickname = 2;
+const theMinimumNumberOfCharactersInThePassword = 8;
+
+const nameSyntax = {
+	valueInput: document.querySelector("#username"),
+	errorName: document.querySelector(".error-name"),
+	iconCheck: document.querySelector(".icon-user-name-correct"),
+	iconX: document.querySelector(".icon-user-name-incorrect"),
+	InputIsCorrect: false,
+};
+const passwordSyntax = {
+	valueInput: document.querySelector("#password"),
+	errorName: document.querySelector(".error-password"),
+	iconCheck: document.querySelector(".icon-password-correct"),
+	iconX: document.querySelector(".icon-password-incorrect"),
+	InputIsCorrect: false,
+};
+
+const ComparingPasswordsSyntax = {
+	valueInput: document.querySelector("#password2"),
+	errorName: document.querySelector(".error-password2"),
+	iconCheck: document.querySelector(".icon-password2-correct"),
+	iconX: document.querySelector(".icon-password2-incorrect"),
+	InputIsCorrect: false,
+};
+
+const citySyntax = {
+	valueInput: document.querySelector("#city"),
+	errorName: document.querySelector(".error-city"),
+	iconCheck: document.querySelector(".icon-city-correct"),
+	iconX: document.querySelector(".icon-city-incorrect"),
+	InputIsCorrect: false,
+};
+
+const emailSyntax = {
+	valueInput: document.querySelector("#email"),
+	errorName: document.querySelector(".error-email"),
+	iconCheck: document.querySelector(".icon-email-correct"),
+	iconX: document.querySelector(".icon-email-incorrect"),
+	InputIsCorrect: false,
+};
```

Comment:
You could create a function which creates object with the same structure so the code is not replicated 4 times